app/scripts/services/import-modal.js

```diff
@@ -55,6 +55,15 @@
 
         return importService.mergeFile($scope.rootDirectory, mode.value)
           .then(function () {
+            var promises = [];
+
+            $scope.rootDirectory.forEachChildDo(function (file) {
```

Comment:
You have a good point, so I changed the strategy to avoid saving those non-imported unsaved files. Now every file from the zip is being saved right after it has been imported.